gcs/src/gcs_group.cpp

```diff
@@ -1137,15 +1148,19 @@ group_find_ist_donor (const gcs_group_t* const group,
         return -1;
     }
 
+    // We should use ist_seqno - safety_gap (instead of ist_seqno)
+    // during the search of the donor node to avoid situations when
+    // seqno on the donor node has changed so much that IST is not
+    // possible and we need to run SST:
     if (str_len) {
         // find ist donor by name.
         idx = group_find_ist_donor_by_name_in_string(
-            group, joiner_idx, str, str_len, ist_seqno, status);
+            group, joiner_idx, str, str_len, ist_seqno - safety_gap, status);
```

Comment:
Can you help me understand this safety_gap semantics.
(As I know this is like considering that donor may make a progress by the time request are resolved 
 but it looks bit more complex that this.)

Also, what if we leave this heuristic logic as before.
It has been working fine in most cases except few for which we now have a solution.